purchase_requisition_line_description/models/purchase_requisition_line.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseRequisitionLine(models.Model):
+    _inherit = "purchase.requisition.line"
+
+    name = fields.Text(string="Description")
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        # to format name into: "[product_code] product_name"
+        res = super()._onchange_product_id()
+        desc = ""
+        if self.product_id:
+            product_lang = self.product_id.with_context(
+                lang=self.requisition_id.purchase_ids.partner_id.lang,
+                partner_id=self.requisition_id.purchase_ids.partner_id.id,
+            )
+            vendor_data = product_lang.seller_ids.filtered(
+                lambda a: a.product_id == self.product_id
+                and (a.product_name
+                or a.product_code)
+            )[0]
```

Comment:
This will probably raise an error if the recordset is empty. You can instead use `fields.first` to safely select first occurrence in a recordset.
Moreover, as this is an OCA module, IMO we should respect the default behaviour of Odoo from the purchase module and only use the display name: https://github.com/odoo/odoo/blob/12.0/addons/purchase/models/purchase.py#L595-L597